di-ipv-cimit-stub/lambdas/stub-management/src/main/java/uk/gov/di/ipv/core/stubmanagement/service/impl/UserServiceImpl.java

```diff
@@ -65,11 +66,10 @@ public void addUserCis(String userId, List<UserCisRequest> userCisRequest) {
 
     @Override
     public void updateUserCis(String userId, List<UserCisRequest> userCisRequest) {
-        if (userCisRequest.stream().allMatch(user -> StringUtils.isEmpty(user.getCode()))) {
-            throw new BadRequestException("User's CI Code cannot be null in all CIs");
-        }
+        checkCICodes(userCisRequest);
         List<CimitStubItem> cimitStubItems = cimitStubService.getCimitStubItems(userId);
         if (!cimitStubItems.isEmpty()) {
```

Comment:
This will silently ignore the list of CIs if there are none on the table.
This could just be: delete all CIs for the user and then insert the supplied CIs - there can only be a single user per PUT or POST.

Reply:
I believe both options will yield the same result. 1-) Here, we are deleting all records except for the CI codes in the incoming request CI codes, and then performing an update. 2-) We delete all CIs for this userId and then inserted all request CIs. Assuming it's a PUT operation, the first option seemed more suitable to me initially, but as you suggested, I'm adjusting the code according to the second option. :)

Reply:
Done.